File: app/src/main/java/ardjomand/leonardo/nutrimeal/users/UsersContract.java
package ardjomand.leonardo.nutrimeal.users;

import ardjomand.leonardo.nutrimeal.data.pojos.User;

interface UsersContract {

    interface View {

        void addUser(User user);

        void updateUser(User user);

        void showEmptyUser();

        void showError();
    }

    interface Presenter {

        void setView(View view);

        void subscribeToUsersUpdates();

        void unsubscribeFromUsersUpdates();
    }
}

File: app/src/main/java/ardjomand/leonardo/nutrimeal/users/UsersContract.kt
package ardjomand.leonardo.nutrimeal.users

import ardjomand.leonardo.nutrimeal.data.pojos.User

internal interface UsersContract {

    interface View {

        fun addUser(user: User)

        fun updateUser(user: User)

        fun showEmptyUser()

        fun showError()
    }

    interface Presenter {

        fun setView(view: View?)

        fun subscribeToUsersUpdates()

        fun unsubscribeFromUsersUpdates()
    }
}
